### PR TITLE
feat: Add command line/environment flag for commonConfig 

### PR DESCRIPTION
**Pull request by FelixTing**
fix: #485 

Hold until https://github.com/edgexfoundry/device-sdk-c/pull/484 is merged

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

**Non-secure mode testing**

1. From the latest compose builder run `make run no-secty`.
2. Set the environment variable for non-secure testing `export EDGEX_SECURITY_SECRET_STORE=false`
3. Run device-random example (/src/c/examples/random) with `-cc=../commonconfiguration.yaml` argument.
4. The device-random service should bootstrap successfully without having the following error
```
level=ERROR ts=2023-10-06T04:34:11Z app=device-random msg="Unable to load common config file: Unable to open configuration file"
Error: 1: Unable to open configuration file
```
5. The configuration can also be verified through the REST API: http://localhost:59999/api/v3/config

**Secure mode testing**

1. From the latest compose builder, modify add-security.yml to add secret store token and registry ACL role for device-random, and then run `make run`.
2. Set the environment variable for secure testing `export EDGEX_SECURITY_SECRET_STORE=true`
3. Run device-random example (/src/c/examples/random) with `-cc=../commonconfiguration.yaml` argument.
4. The device-random service should bootstrap successfully without the following error
```
level=ERROR ts=2023-10-06T04:34:11Z app=device-random msg="Unable to load common config file: Unable to open configuration file"
Error: 1: Unable to open configuration file
```

### CODE REVIEW

File: src/c/examples/bitfields/res/configuration.yaml [hunk 1]
```diff
@@ -6,21 +6,17 @@ Service:
...
 
-Clients:
-  core-metadata:
-    Host: localhost
-    Port: 59881
+# uncomment when running from command-line in hybrid mode without Config Provider
+#Clients:
+#  core-metadata:
+#    Host: localhost
+#    Port: 59881
```
Comment (lenny-intel) [inline src/c/examples/bitfields/res/configuration.yaml:13]: This is not needed if you also implement the -d/--dev flag to set the automatically set the different Hosts for hybrid mode. :-)
See these PRs in go.
https://github.com/edgexfoundry/go-mod-bootstrap/pull/516
https://github.com/edgexfoundry/go-mod-bootstrap/pull/541
https://github.com/edgexfoundry/go-mod-bootstrap/pull/601

File: src/c/examples/bitfields/res/configuration.yaml [hunk 2]
```diff
@@ -6,21 +6,17 @@ Service:
...
   DevicesDir: ./res/devices
 
 MessageBus:
-  Protocol: redis
-  Host: localhost
-  Port: 6379
-  Type: redis
-  AuthMode: usernamepassword
-  SecretName: redisdb
+
```
Comment (lenny-intel) [inline src/c/examples/bitfields/res/configuration.yaml:20]: remove white space

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: src/c/examples/bitfields/res/configuration.yaml (hunk 2 above)
Comment (FelixTing) [reply src/c/examples/bitfields/res/configuration.yaml]: removed

File: src/c/examples/bitfields/res/configuration.yaml (hunk 1 above)
Comment (FelixTing) [reply src/c/examples/bitfields/res/configuration.yaml:13]: Thank you @lenny-intel. I have created an issue for this. https://github.com/edgexfoundry/device-sdk-c/issues/487

File: src/c/service.c [hunk 3]
```diff
@@ -999,10 +1027,15 @@ void devsdk_service_start (devsdk_service_t *svc, iot_data_t *driverdfls, devsdk
   }
   else
   {
-    edgex_device_parseClients (svc->logger, iot_data_string_map_get (config_file, "Clients"), &svc->config.endpoints);
+    edgex_device_parseClients (svc->logger, iot_data_string_map_get (deviceservices_config, "Clients"), &svc->config.endpoints);
   }
 
   iot_data_free (config_file);
+  if (common_config_file)
```
Comment (iain-anderson) [inline src/c/service.c:1034]: `iot_data_free()` follows the `free()` convention of allowing a NULL parameter, so you don't need to protect it with an if

Comment (FelixTing) [reply src/c/service.c]: updated

File: src/c/service.c [hunk 4]
```diff
@@ -854,7 +857,33 @@ void devsdk_service_start (devsdk_service_t *svc, iot_data_t *driverdfls, devsdk
...
+  {
+    common_config_file = edgex_device_loadConfig (svc->logger, svc->commonconffile, err);
+    if (err->code)
+    {
+      iot_log_error (svc->logger, "Unable to load common config file: %s", err->reason);
+      return;
+    }
+    iot_data_t *allservices_config_key = iot_data_alloc_string("all-services", IOT_DATA_REF);
+    iot_data_t *deviceservices_config_key = iot_data_alloc_string("device-services", IOT_DATA_REF);
+    iot_data_t *allservices_config = iot_data_map_get_map(common_config_file, allservices_config_key);
```
Comment (iain-anderson) [inline src/c/service.c:873]: You can just use `iot_data_string_map_get_map (common_config_file, "all-services");`
Also that string is #defined in consul.c, maybe move the definition to eg api.h?

Comment (FelixTing) [reply src/c/service.c]: modified

**Comment by FelixTing** [comment]
Hi @lenny-intel, this PR cannot be merged due to the change request. Could you please approve this PR if you are happy with the changes?

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**